addons/portal/wizard/portal_wizard.py

```diff
@@ -174,7 +174,8 @@ def action_revoke_access(self):
         # remove the user from the portal group
         if user_sudo and user_sudo.has_group('base.group_portal'):
             # if user belongs to portal only, deactivate it
-            if len(user_sudo.groups_id) <= 1:
+            public_user_groups = user_sudo._public_user_group_whitelist()
+            if (public_user_groups + group_portal) == user_sudo.groups_id:
```

Comment:
Why not simply detect if belonging to portal group and remove the rest the code?

```suggestion
            if group_portal & user_sudo.groups_id:
```

Reply:
Hello :wave: 

The intent of this code is to prevent removing users if they are more than just portal users, this suggestion would change the intended behavior.

I understand the point of the PR currently is simply to make the set of groups that would result in archival expendable (?) Which might be ok

Reply:
I just changed the commit message. I hope is more clear now :smile: Thanks for the review!

Reply:
@thiryr @chienandalu but the existing condition is not real, as Odoo assigns by default other groups to portal users when you select them in the configuration (like the B2B/B2C one).

As user type (Internal, Portal, Public) are mutual exclusive, doing such comparison is enough for determining if a user is a portal one or not. No need of complicated inheritable methods to determine which ones are allowed or not IMO.